### PR TITLE
don't check the first version in the list; don't check auth if its no…

### PR DESCRIPTION
**Pull request by odise**
…t configured

### CODE REVIEW

File: pkg/mirror/service.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func (p *pullThroughMirror) RetrieveProviderArchive(ctx context.Context, provide
 }
 
 func (p *pullThroughMirror) upstreamSha256Sums(ctx context.Context, provider *core.Provider, versions *core.ProviderVersions) (*core.Sha256Sums, error) {
-	if len(versions.Versions) == 0 || len(versions.Versions[0].Platforms) == 0 {
+	if len(versions.Versions) == 0 {
 		return nil, errors.New("core.ProviderVersions doesn't contain any platforms")
```
Comment (oliviermichaelis) [inline pkg/mirror/service.go:165]: Can we change the error message to `"core.ProviderVersions doesn't contain any versions"` please?

**Review by oliviermichaelis — COMMENTED** [review]
One minor request, otherwise it looks good to me! :+1:

File: pkg/mirror/service.go (hunk 1 above)
Comment (odise) [reply pkg/mirror/service.go]: Done.

**Review by oliviermichaelis — APPROVED** [review]
LGTM :+1:

**Pull request merged**